external-crates/move/crates/move-compiler/src/parser/ast.rs

```diff
@@ -301,16 +306,65 @@ pub struct Constant {
 // Types
 //**************************************************************************************************
 
+// MName = Name <TyArgs>
+//       | Leading <TyArgs> :: Name
+//       | Leading :: Name <TyArgs>
+//       | Leading :: Name <TyArgs> :: Name
+//       | Leading :: Name :: Name <TyArgs>
+//       | Leading :: Name :: Name <TyArgs> :: Name
+//
+// VName = MName <TyArgs> :: Name
+
 // A ModuleAccess references a local or global name or something from a module,
 // either a struct type or a function.
+// The boolean flag indicates if a name was pushed on after tyarg parsing, which we use in
+// expansion to determine if the variant form was valid.
+// FIXME(cswords): We should eventually rewrite all of this to be `NamePath`s that hold tyargs:
+//    pub enum NamePath {
+//        Base { name: Name, tyargs: Option<Vec<Type>> },
+//        Leading { name: LeadingNameAccess },
+//        Ext { base: Box<NamePath>, name: Name, tyargs: Option<Vec<Type>> },
+//    }
+
+// A single name with optional type arguments that may be a macro call.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PathEntry {
+    pub name: Name,
+    pub tyargs: Option<Spanned<Vec<Type>>>,
+    pub is_macro: Option<Loc>,
+}
+
+// A path root.
+// For now these should hever have tyargs or macro call set (though the type arguments will be
+// used for enums).
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RootPathEntry {
+    pub name: LeadingNameAccess,
+    pub tyargs: Option<Spanned<Vec<Type>>>,
+    pub is_macro: Option<Loc>,
```

Comment:
I didn't see any tests for this? I figure based on conversations that it was a later edition, but would still be good to test

Reply:
Added a test.